src/cryptonote_core/service_node_list.cpp

```diff
@@ -145,6 +146,12 @@ namespace service_nodes
 
         process_block(block, txs);
       }
+
+      if (start_height == m_state.height)
+      {
+        MERROR("Unexpected state height did not change after processing blocks, height is: " << start_height);
+        return;
+      }
```

Comment:
Is there a legitimate case where this state can actually happen, or should this just be an assert?

Reply:
There was a reported case where someone's node took over 15 mins and didn't progress with pop-blocks, so I'm assuming there's some weird edge case that I couldn't track down.